File: src/main/java/com/vocalclerk/api/dtos/CategoryDto.java
package com.vocalclerk.api.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

/**
 * DTO for transferring category data to and from the API.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CategoryDto {
    private UUID id;
    private String name;
    private String color;
    private String icon;
    private boolean isDefault;
    private int taskCount;
}

File: backend/VocalClerk.Application/DTOs/CategoryDto.cs
namespace VocalClerk.Application.DTOs;

public class CategoryDto
{
    public long Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Color { get; set; } = "#808080";
    public string Icon { get; set; } = "📁";
    public bool IsDefault { get; set; }
    public DateTime CreatedAt { get; set; }
}

public class CreateCategoryDto
{
    public string Name { get; set; } = string.Empty;
    public string Color { get; set; } = "#808080";
    public string Icon { get; set; } = "📁";
    public bool IsDefault { get; set; } = false;
}

public class UpdateCategoryDto
{
    public string Name { get; set; } = string.Empty;
    public string Color { get; set; } = "#808080";
    public string Icon { get; set; } = "📁";
    public bool IsDefault { get; set; } = false;
}
